Sources/SymbolKit/SymbolGraph/Symbol/Symbol+KindIdentifier.swift

```diff
@@ -14,87 +14,61 @@ extension SymbolGraph.Symbol {
     /**
      A unique identifier of a symbol's kind, such as a structure or protocol.
      */
-    public enum KindIdentifier: Equatable, Hashable, Codable, CaseIterable {
-        case `associatedtype`
-        case `class`
-        case `deinit`
-        case `enum`
-        case `case`
-        case `func`
-        case `operator`
-        case `init`
-        case method
-        case property
-        case `protocol`
-        case `struct`
-        case `subscript`
-        case typeMethod
-        case typeProperty
-        case typeSubscript
-        case `typealias`
-        case `var`
+    public struct KindIdentifier: Equatable, Hashable, Codable {
+        public let identifier: String
 
-        case module
-
-        case unknown
-
-        /// A string that uniquely identifies the symbol kind.
-        ///
-        /// If the original kind string was not recognized, this will return `"unknown"`.
-        public var identifier: String {
-            switch self {
-            case .associatedtype: return "associatedtype"
-            case .class: return "class"
-            case .deinit: return "deinit"
-            case .enum: return "enum"
-            case .case: return "enum.case"
-            case .func: return "func"
-            case .operator: return "func.op"
-            case .`init`: return "init"
-            case .method: return "method"
-            case .property: return "property"
-            case .protocol: return "protocol"
-            case .struct: return "struct"
-            case .subscript: return "subscript"
-            case .typeMethod: return "type.method"
-            case .typeProperty: return "type.property"
-            case .typeSubscript: return "type.subscript"
-            case .typealias: return "typealias"
-            case .var: return "var"
-            case .module: return "module"
-            case .unknown: return "unknown"
-            }
+        private init(rawIdentifier: String) {
```

Comment:
Is there any value in making this public so clients can create new symbol kinds without updating SymbolKit itself?

Reply:
Oh, it looks like we have a public initializer below?

Reply:
The public initializer first checks against the known symbol kinds, then returns the original kind identifier string unchanged if it doesn't match any of the known ones. This at least can keep down on string allocations that are stored in the final symbol graph, if it uses known symbol kinds. In the end, it does defer to this `rawIdentifier` initializer, either via a static property or by storing the unknown kind string.